packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/CameraApiImpl.java

```diff
@@ -0,0 +1,480 @@
+    Integer videoBitrate =
+        (settings.getVideoBitrate() == null) ? null : settings.getVideoBitrate().intValue();
+    Integer audioBitrate =
+        (settings.getAudioBitrate() == null) ? null : settings.getAudioBitrate().intValue();
+    ResolutionPreset resolutionPreset =
+        CameraUtils.resolutionPresetFromPigeon(settings.getResolutionPreset());
+
+    camera =
+        new Camera(
+            activity,
+            flutterSurfaceTexture,
+            new CameraFeatureFactoryImpl(),
+            dartMessenger,
+            cameraProperties,
+            new Camera.VideoCaptureSettings(
+                resolutionPreset, settings.getEnableAudio(), fps, videoBitrate, audioBitrate));
+
+    return flutterSurfaceTexture.id();
+  }
+
+  // We move catching CameraAccessException out of onMethodCall because it causes a crash
+  // on plugin registration for sdks incompatible with Camera2 (< 21). We want this plugin to
+  // to be able to compile with <21 sdks for apps that want the camera and support earlier version.
+  @SuppressWarnings("ConstantConditions")
+  private <T> void handleException(Exception exception, Messages.Result<T> result) {
+    if (exception instanceof CameraAccessException) {
+      result.error(exception);
+      return;
+    }
+
+    // CameraAccessException can not be cast to a RuntimeException.
+    throw (RuntimeException) exception;
+  }
+
+  private void handleException(Exception exception, Messages.VoidResult result) {
+    if (exception instanceof CameraAccessException) {
+      result.error(exception);
+      return;
+    }
+
+    // CameraAccessException can not be cast to a RuntimeException.
+    throw (RuntimeException) exception;
+  }
+
+  @NonNull
+  @Override
+  public List<Messages.PlatformCameraDescription> getAvailableCameras() {
+    if (activity == null) {
+      return Collections.emptyList();
+    }
+    try {
+      return CameraUtils.getAvailableCameras(activity);
+    } catch (CameraAccessException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void create(
+      @NonNull String cameraName,
+      @NonNull Messages.PlatformMediaSettings settings,
+      @NonNull Messages.Result<Long> result) {
+    if (camera != null) {
+      camera.close();
+    }
+
+    cameraPermissions.requestPermissions(
+        activity,
+        permissionsRegistry,
+        settings.getEnableAudio(),
+        (String errCode, String errDesc) -> {
+          if (errCode == null) {
+            try {
+              result.success(instantiateCamera(cameraName, settings));
+            } catch (Exception e) {
+              handleException(e, result);
+            }
+          } else {
+            result.error(new Messages.FlutterError(errCode, errDesc, null));
+          }
+        });
+  }
+
+  @Override
+  public void initialize(
+      @NonNull Long cameraId,
+      @NonNull Messages.PlatformImageFormatGroup imageFormat,
+      @NonNull Messages.VoidResult result) {
```

Comment:
All of the uses of `result` appear to be synchronous here, which means this method does not need to be `@async` in the Pigeon definition.

While the main improvement Pigeon provides is type safety, not having to track result objects when the implementation doesn't require it is another significant safety win because the compiler guarantees that there is always exactly one reply (as the method channel contract requires) vs. having to make sure there are never codepaths with 0 or >1 calls to result, so methods should only be declared `@sync` when they actually have to be.

Reply:
I figured I may as well handle these as soon as I'm able. Let me know if the new state is any better.